src/links/v1/providers.json

```diff
@@ -65,9 +65,9 @@
       "id": "mp",
       "attributes": {
         "name": "materialsproject.org",
-        "description": "",
-        "base_url": null,
-        "homepage": null
+        "description":  "The Materials Project: A database of computed materials properties for the discovery and design of novel materials.",
+        "base_url": "https://optimade.materialsproject.org",
```

Comment:
This does not seem to be an index meta-db as is required (which is also the case for @ml-evs's `odbx` server).
Since it's better with something instead of nothing, it's fine for me, but since we are explicitly requiring an index meta-db in the spec for this `providers.json`, we should at some point be a bit more clear cut. Maybe when `v1` is released?

Reply:
See both [here](https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#index-meta-database) and [here](https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#database-provider-specific-namespace-prefixes) in the spec for the "requirement".

Reply:
Hmm, ok. I think we need docs on the python tools because I have no clue how to deploy/find this index server once it's deployed. We can leave this PR up till I figure that out and re-deploy.

Reply:
Right. It's actually started with the `docker-compose.yml` as well, but yeah sure. It all depends on your deployment situation in the end. But let's take that discussion to another place (maybe slack?).

Reply:
I think this needs rewording in the spec tbh, if the index meta-db is entirely optional (which was my understanding) then this providers repo should also allow a non-index-db to be listed as the sole database provided by that provider. Otherwise, we should make the index meta-db a hard requirement.

I see from your links that the "note" part implies that providers in this MUST point to an index.

Reply:
@CasperWA As far as I can see in the specification if you have only one database as a provider you don't need to have an index meta-database but you can use the URL for the "real" database. 
I thought this is the main purpose of the database discovery and the index meta-databases. By using the parent/child relation you can create an arbitrary deep tree when the leaves are the actual databases. If you have a single database that a leaf.

Reply:
Another thing @shyamd. Here you are using https://optimade.materialsproject.org as the URL, but  http://optimade.materialsproject.org:443/v0/info in the actual deployment which leads to `400 Bad Request`. As far as I know, this is due to trying to access an `https` server with the `http` protocol

Reply:
(Sorry for stealing your PR @shyamd)
There are several reasons for and against having the index meta-db between this list and the final database.
Originally, this was implemented to provide an easy gateway (of sorts) for providers with multiple databases to serve them clearly. Within the consortium (according to the spec) we only support two-layered "trees" in the term of a provider's database layout: The single parent is the index meta-db and any other database is a child of it. However, that's not to say that the spec does not allow it, rather within the list of providers by the consortium, this is an additional requirement.

To accommodate the providers with only a single database (various editions of *cod, odbx (at the moment), materials project, etc.) the index-meta db has the `default` property under it's `/info`-endpoint. One can then supply the `id` for the database here and a client can find it's corresponding URL under the `/links` endpoint. This means all files needed for an index meta-db can be static. Furthermore, it should be up to client to decide what to do, not the providers.

In case of errors, the index meta-db also acts as a fallback (see [here](https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#http-response-status-codes)).

The index meta-db let's all the consortiums providers be at the "same level", in the sense of how to deliver the data. Because a client is first faced with the data-empty index meta-db it has the potential to be more intelligible about what a provider offers of data, quantities and content. It also provides a clear possibility for more compartmentalization of data into separate databases from the perspective of a provider (e.g., if one has had the fortune to separate each scientific project into a separate database, one can use OPTIMADE to quickly peruse all projects).

In this sense I think it is a compromise that tries to accommodate as many different flavors of providers as possible, while keeping it as fair as possible.

Reply:
> Another thing @shyamd. Here you are using https://optimade.materialsproject.org as the URL, but http://optimade.materialsproject.org:443/v0/info in the actual deployment which leads to `400 Bad Request`. As far as I know, this is due to trying to access an `https` server with the `http` protocol

This is a minor issue of the landing page.
The url should indeed be as @shyamd has written it, since it is the version-less base URL we want.

Reply:
Yeah, I forgot to set `base_url`, so it's using the URL from the request, which gets the SSL certificate from Cloudflare.

Reply:
> > Another thing @shyamd. Here you are using https://optimade.materialsproject.org as the URL, but http://optimade.materialsproject.org:443/v0/info in the actual deployment which leads to `400 Bad Request`. As far as I know, this is due to trying to access an `https` server with the `http` protocol
> 
> This is a minor issue of the landing page.
> https://optimade.materialsproject.org/v0/info works fine.
> The url should indeed be as @shyamd has written it, since it is the version-less base URL we want.

The issue is not a version-less or not but rather the first part `http://optimade.materialsproject.org:443`. Try to click on any link on https://optimade.materialsproject.org

Reply:
Agreed @fekad. And @shyamd already posted the underlying reason. Luckily the landing page(s) are simply a help for a human not necessary for a client. For a client this all works great :)

Reply:
For the somewhat lengthy discussion on this, see Materials-Consortia/OPTIMADE#46, Materials-Consortia/OPTIMADE#63. However, I think the main design was agreed on during a Skype call. My recollection is that the requirement that all providers must supply an index database isn't meant to be in the specification, because it isn't strictly a requirement on a conformant implementation. It was however meant to be part of a Materials-Consortia *policy* that a provider has to supply an index meta-database to get indexed in our provides database.

As for the motivation of that policy, my recollection of a good argument for it was that it is a good thing to force providers to supply a one-level indirection. This avoids them getting into trouble later when they realize they want to provide a second dataset. A provider who had submitted a URL to a single database to the providers list would at that point have to choose between two bad options: 
- Changing the URL that previously pointed to a specific database to now suddenly point at an index meta-database (which could break things for others)
- Changing their entry in the providers database to a different URL that points to the index meta-database. However, if they had chosen an elegant URL (e.g.,  "www.example.com/optimade" or "optimade.example.com"), they may not want to do that.

Also note that one can also set up an index-meta database by cloning the providers repository and edit a few files. We used to have some documentation on that in the README, but it seems to have been lost when I re-designed the setup for netlify. I should bring those instructions back.

Reply:
These instructions are indeed already on their way back (see your own PR #21).

Reply:
@CasperWA you mean #18. But, oh, right, they just didn't make it in yet. We should get that merged.

Reply:
Thanks and sorry to bother, I didn't know about it I just tried to follow the specification :)

Reply:
No worries @fekad! We need fresh eyes. And please create issues (or better PRs) for sections in the spec that are unclear. It would good to get as much sorted as possible before the major release.

Reply:
@fekad, @CasperWA I couldn't agree more; the feedback on what is unclear is very valuable. This whole discussion reminds me that we should put our policy in writing *somewhere*, probably in the README here in the providers repository.